src/gpu_sysman.c

```diff
@@ -1190,11 +1190,13 @@ static bool gpu_mems_bw(gpu_device_t *gpu) {
 
 /* set frequency metric labels based on its properties, return true for success
  */
-static bool set_freq_labels(zes_freq_handle_t freq, metric_t *metric) {
+static bool set_freq_labels(zes_freq_handle_t freq, metric_t *metric,
+                            double *maxfreq) {
   zes_freq_properties_t props;
   if (zesFrequencyGetProperties(freq, &props) != ZE_RESULT_SUCCESS) {
     return false;
   }
+  *maxfreq = props.max;
```

Comment:
In a perfect world a "set" function shouldn't get. :)

Reply:
I guess it could be renamed to `set_freq_metric_labels()`, but then same should be done also for `set_mem_labels()` and IMHO it's enough that comments for both of those functions already state that they are for setting the metric (not sysman handle)?